Lib/profiling/sampling/cli.py

```diff
@@ -0,0 +1,705 @@
+    "gecko": GeckoCollector,
+}
+
+
+def _parse_mode(mode_string):
+    """Convert mode string to mode constant."""
+    mode_map = {
+        "wall": PROFILING_MODE_WALL,
+        "cpu": PROFILING_MODE_CPU,
+        "gil": PROFILING_MODE_GIL,
+    }
+    return mode_map[mode_string]
+
+
+def _run_with_sync(original_cmd, suppress_output=False):
+    """Run a command with socket-based synchronization and return the process."""
+    # Create a TCP socket for synchronization with better socket options
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sync_sock:
+        # Set SO_REUSEADDR to avoid "Address already in use" errors
+        sync_sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        sync_sock.bind(("127.0.0.1", 0))  # Let OS choose a free port
+        sync_port = sync_sock.getsockname()[1]
+        sync_sock.listen(1)
+        sync_sock.settimeout(_SYNC_TIMEOUT)
+
+        # Get current working directory to preserve it
+        cwd = os.getcwd()
+
+        # Build command using the sync coordinator
+        target_args = original_cmd[1:]  # Remove python executable
+        cmd = (
+            sys.executable,
+            "-m",
+            "profiling.sampling._sync_coordinator",
+            str(sync_port),
+            cwd,
+        ) + tuple(target_args)
+
+        # Start the process with coordinator
+        # Suppress stdout/stderr if requested (for live mode)
+        popen_kwargs = {}
+        if suppress_output:
+            popen_kwargs["stdin"] = subprocess.DEVNULL
+            popen_kwargs["stdout"] = subprocess.DEVNULL
+            popen_kwargs["stderr"] = subprocess.DEVNULL
+
+        process = subprocess.Popen(cmd, **popen_kwargs)
+
+        try:
+            # Wait for ready signal with timeout
+            with sync_sock.accept()[0] as conn:
+                ready_signal = conn.recv(_RECV_BUFFER_SIZE)
+
+                if ready_signal != _READY_MESSAGE:
+                    raise RuntimeError(
+                        f"Invalid ready signal received: {ready_signal!r}"
+                    )
+
+        except socket.timeout:
+            # If we timeout, kill the process and raise an error
+            if process.poll() is None:
+                process.terminate()
+                try:
+                    process.wait(timeout=_PROCESS_KILL_TIMEOUT)
+                except subprocess.TimeoutExpired:
+                    process.kill()
+                    process.wait()
+            raise RuntimeError(
+                "Process failed to signal readiness within timeout"
+            )
+
+        return process
+
+
+def _add_sampling_options(parser):
+    """Add sampling configuration options to a parser."""
+    sampling_group = parser.add_argument_group("Sampling configuration")
+    sampling_group.add_argument(
+        "-i",
+        "--interval",
+        type=int,
+        default=100,
+        metavar="MICROSECONDS",
+        help="Sampling interval",
```

Comment:
The autogenerated `--help` start lowercase:

```
  -h, --help            show this help message and exit
```

It'd be consistent to put all of these lowercase too, but I don't really...

```suggestion
        help="sampling interval",
```

Reply:
I wasn't very clear here :)

I meant: it'd be consistent to make _every single_ `help=` start lowercase, but I don't really know if it's worth it...

Reply:
Oh, will try to fold this fix in a different PR then. Apologies for the confusion